catroid/src/androidTestCreateatschool/java/org/catrobat/catroid/createatschool/uiespresso/ui/activity/RestrictedLoginTest.java

```diff
@@ -0,0 +1,253 @@
+		Intents.release();
+	}
+
+	@Test
+	public void restrictedLoginDialogNotShownAfterConfirmationTest() {
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(baseActivityTestRule.getActivity());
+		assertTrue("Token has been set but shouldn't be!", sharedPreferences.getBoolean(Constants.RESTRICTED_LOGIN_ACCEPTED, false));
+	}
+
+	@Test
+	public void loginDialogAppearsAfterLogoutTest() {
+		setFakeToken(baseActivityTestRule.getActivity(), true);
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(baseActivityTestRule.getActivity());
+		sharedPreferences.edit().putBoolean(Constants.CREATE_AT_SCHOOL_USER, true).apply();
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(doesNotExist());
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(doesNotExist());
+
+		openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getTargetContext());
+		onView(withText(R.string.main_menu_logout))
+				.perform(click());
+
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void loginWithOtherThanCreateAtSchoolUserTest() {
+		String testUserName = "testUser" + System.currentTimeMillis();
+		String testPassword = "password";
+		String testUserMail = testUserName + "@catrob.at";
+		createValidUserWithCredentials(baseActivityTestRule.getActivity(), testUserName, testPassword, testUserMail);
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onView(withId(R.id.dialog_login_username))
+				.perform(clearText(), typeText(testUserName), closeSoftKeyboard());
+
+		onView(withId(R.id.dialog_login_password))
+				.perform(clearText(), typeText(testPassword), closeSoftKeyboard());
+
+		onView(allOf(withText(R.string.login), withResourceName("button1")))
+				.perform(click());
+
+		onView(withText(R.string.error_no_nolb_user))
+				.check(matches(isDisplayed()));
+
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(baseActivityTestRule.getActivity());
+		assertEquals("Token has been set but shouldn't be!", Constants.NO_TOKEN, sharedPreferences.getString(Constants.TOKEN, Constants.NO_TOKEN));
+		assertEquals("Username has been set but shouldn't be!", Constants.NO_USERNAME, sharedPreferences.getString(Constants.USERNAME, Constants.NO_USERNAME));
+	}
+
+	private void setFakeToken(Context context, boolean valid) {
+		String token = valid ? VALID_FAKE_TOKEN : INVALID_FAKE_TOKEN;
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		sharedPreferences.edit().putString(Constants.TOKEN, token).apply();
+	}
+
+	private void createValidUserWithCredentials(Context context, String testUser, String testPassword, String testEmail) {
+		try {
```

Comment:
let this function throw the webconnection exception to the test.
so if that breaks it at least shows the right exception and not the test failing due to some viewAssertion. this would make debugging easier / clear to spot what exactly is wrong.